### PR TITLE
Throw Stripe error on invalid JSON

### LINKED ISSUE #588: Method "LiveStripeResponseGetter.handleApiError()" throws unhandled exception "com.google.gson.JsonSyntaxException"
Java 8.0
stripe-java library 5.31.0 but the method implementation is currently the very same.

```
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BEGIN_OBJECT but was STRING at line 1 column 1 path $
    at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:224)
    at com.google.gson.Gson.fromJson(Gson.java:887)
    at com.google.gson.Gson.fromJson(Gson.java:852)
    at com.google.gson.Gson.fromJson(Gson.java:801)
    at com.google.gson.Gson.fromJson(Gson.java:773)
    at com.stripe.net.LiveStripeResponseGetter.handleAPIError(LiveStripeResponseGetter.java:663)
    at com.stripe.net.LiveStripeResponseGetter._request(LiveStripeResponseGetter.java:508)
    at com.stripe.net.LiveStripeResponseGetter.request(LiveStripeResponseGetter.java:72)
    at com.stripe.net.APIResource.request(APIResource.java:169)
    at com.stripe.model.Plan.retrieve(Plan.java:219)
    at com.stripe.model.Plan.retrieve(Plan.java:181)
```

My suggestion is that this method should catch any thrown "com.google.gson.JsonSyntaxException" and handle it properly as expected.

### PR DESCRIPTION
**Pull request by ob-stripe**
r? @remi-stripe 
cc @stripe/api-libraries 

Throw a `com.stripe.exception.ApiException` when encountering invalid JSON. This is similar to what we do in other libraries, e.g. in stripe-ruby: https://github.com/stripe/stripe-ruby/blob/master/lib/stripe/stripe_client.rb#L285-L286.

Unfortunately, this is difficult to test because this code is in private static method, and AFAIK it's not possible to reproduce this issue via stripe-mock.

Fixes #588 (well, it doesn't fix the root cause, but that cause is almost definitely a backend or network issue of some sort, not a library bug)

### CODE REVIEW

**Comment by ob-stripe** [comment]
You're right, I updated the PR to handle all 4 places where we try and parse JSON (error / not error, API / OAuth).

I've also ran a couple manual tests with a dummy server:

Not error:
```
17:04:22.886 [DEBUG] [TestEventLogger]     com.stripe.exception.ApiException: Invalid response object from API: Boo!. (HTTP response code was 200)
17:04:22.886 [DEBUG] [TestEventLogger]         at com.stripe.net.LiveStripeResponseGetter.raiseGeneralApiError(LiveStripeResponseGetter.java:711)
17:04:22.887 [DEBUG] [TestEventLogger]         at com.stripe.net.LiveStripeResponseGetter.staticRequest(LiveStripeResponseGetter.java:540)
```

Error:
```
17:03:20.544 [DEBUG] [TestEventLogger] com.stripe.functional.ChargeTest > testRetrieve FAILED
17:03:20.544 [DEBUG] [TestEventLogger]     com.stripe.exception.ApiException: Invalid response object from API: Boo!. (HTTP response code was 400)
17:03:20.545 [DEBUG] [TestEventLogger]         at com.stripe.net.LiveStripeResponseGetter.raiseGeneralApiError(LiveStripeResponseGetter.java:711)
17:03:20.545 [DEBUG] [TestEventLogger]         at com.stripe.net.LiveStripeResponseGetter.handleApiError(LiveStripeResponseGetter.java:722)
17:03:20.545 [DEBUG] [TestEventLogger]         at com.stripe.net.LiveStripeResponseGetter.staticRequest(LiveStripeResponseGetter.java:533)
```

File: src/main/java/com/stripe/net/LiveStripeResponseGetter.java [hunk 1]
```diff
@@ -690,11 +702,25 @@ private static StripeResponse getMultipartStripeResponse(
 
   }
 
+  private static void raiseGeneralApiError(String responseBody, int responseCode, String requestId)
```
[the remark this answers is not in the document]
Comment (ob-stripe) [reply src/main/java/com/stripe/net/LiveStripeResponseGetter.java]: I reused the name from stripe-ruby, but fair enough :) I'll rename the method.

**Comment by ob-stripe** [comment]
🤔 Would this qualify as a breaking change? This error should never happen anyway, but we know it does occasionally, and maybe some people are currently catching `JsonSyntaxException` :/

@brandur-stripe Any thoughts on this?

**Comment by brandur-stripe** [comment]
> 🤔 Would this qualify as a breaking change? This error should never happen anyway, but we know it does occasionally, and maybe some people are currently catching `JsonSyntaxException` :/
>
> @brandur-stripe Any thoughts on this?

Somewhat ambiguous, but given that people are almost certainly already handling Stripe errors, I'd personally probably release it as non-breaking.

**Comment by ob-stripe** [comment]
Yeah, sounds reasonable. I'll release this as a minor and make a note in the changelog. Thanks Brandur and Remi!

**Comment by ob-stripe** [comment]
Released as 7.6.0.

**Pull request merged**